src/DsaPrinter.cc

```diff
@@ -656,6 +687,42 @@ struct DsaPrinter : public ModulePass {
       if (G->begin() != G->end()) {
         std::string Filename = F.getName().str() + ".mem.dot";
         writeGraph(G, Filename);
+
+        if (DsaColorCallSiteSimDot){
+          const Function *f_caller = &F;
+
+          for (auto &bb : F) {
+            for (auto &instruction : bb) {
+              if (CallInst *callInst = dyn_cast<CallInst>(&instruction)) {
```

Comment:
There are two problems with this code. First we directly exclude `InvokeInst`. More importantly, we ignore the information from `DsaCompleteCallGraph`. The main benefit of this is to that the DsaCallGraph may have resolved indirect calls that LLVM cannot. If you just have the check at line 697 we will exclude indirect calls that are actually direct. There are other places in sea-dsa where we use the `DsaCompleteCallGraph` to extract all callsites in a program. Check https://github.com/seahorn/sea-dsa/blob/master/src/DsaGlobal.cc#L334.

Reply:
```
  cg = call graph returned by DsaCompleteCallGraph
  for (auto it = scc_begin(&cg); !it.isAtEnd(); ++it) {
    auto &scc = *it;
    for (CallGraphNode *cgn : scc) {
      Function *fn = cgn->getFunction();
      if (!fn || fn->isDeclaration() || fn->empty()) {
        continue;
      }
     
      for (auto &callRecord : *cgn) {
     
	llvm::Optional<DsaCallSite> dsaCS = call_graph_utils::getDsaCallSite(callRecord);
	if (!dsaCS.hasValue()) {
	  continue;
	}
        dsaCS.getValue() is what you need 
```